cns/restserver/ipam.go

```diff
@@ -762,6 +783,11 @@ func (service *HTTPRestService) AssignAvailableIPConfigs(podInfo cns.PodInfo) ([
 	numIPsNeeded := len(service.state.ContainerStatus)
 	// Creates a slice of PodIpInfo with the size as number of NCs to hold the result for assigned IP configs
 	podIPInfo := make([]cns.PodIpInfo, numIPsNeeded)
+	// if there are no NCs on the NNC there will be no IPs in the pool so return error
+	if numIPsNeeded == 0 {
+		//nolint:goerr113 // return error
+		return podIPInfo, fmt.Errorf("No NCs found on the NNC so no IPs are in the pool")
```

Comment:
This is exactly why the error should be reified in a var--so it can be reused here (and checked by callers).